route/build/src/init_model_data.f90

```diff
@@ -418,6 +419,9 @@ SUBROUTINE init_state_data(pid, nNodes, comm, ierr, message)
       end if
       if (onRoute(impulseResponseFunc)) then
         do ix = 1,nRch_root
+          ntdh = size(NETOPO_trib(ix)%UH)
+          allocate(RCHFLX_trib(iens,ix)%ROUTE(idxIRF)%QFUTURE_IRF(ntdh), stat=ierr, errmsg=cmessage)
```

Comment:
can combine allocation and initialization like this.
`allocate(RCHFLX_trib(iens,ix)%QFUTURE_IRF(ntdh), source=0._dp, stat=ierr, errmsg=cmessage)`